herbert_core/classes/MPIFramework/@MessagesFilebased/MessagesFilebased.m

```diff
@@ -211,6 +256,21 @@ function clear_messages(obj)
         function val = get.time_to_fail(obj)
             val = obj.time_to_fail_ ;
         end
+        %
+        function is = is_job_canceled(obj)
+            % method verifies if job has been canceled
+            if ~exist(obj.mess_exchange_folder_,'dir')
```

Comment:
can't this just be `isCancelled = ~exist(....)` ?

Reply:
it can, but I believe its other code there. It will return on this but other check should follow if it does not.

Reply:
you are evaluating a boolean to set a boolean variable; then using that boolean variable to conditionally execute a further chunk of code.
It could be simplified to:
```
if exist(....)
    mess = obj.probe_all('any','canceled');
    is = ~isempty(mess)
else
    is = false
end
```
or even
```
is = exist(...) && ~isempty(obj.probe_all('any','canceled'))
```

Reply:
I have changed it for complacency sake, but not sure if the change is for good. If it was absolutely clear what is going on before. It needs some brain processing power to understand what is happening now. And performance here is not an issue at all.

Reply:
the one-liner is a bit cryptic (and possibly wrong). Using a bool comparison to set a bool variable is just ugly.